packages/medusa-plugin-economic/src/services/economic.js

```diff
@@ -84,52 +85,50 @@ class EconomicService extends BaseService {
     }
 
     order.items.forEach((item) => {
-      // For bundles, we create an order line for each item in the bundle
-      if (Array.isArray(item.content)) {
-        item.content.forEach((c) => {
-          const total_amount = c.unit_price * c.quantity * (taxRate + 1)
-
-          order_lines.push({
-            lineNumber: order_lines.length + 1,
-            sortKey: 1,
-            unit: {
-              unitNumber: this.options_.unit_number,
-            },
-            product: {
-              productNumber: c.product.sku,
-            },
-            quantity: c.quantity,
-            // Do we include taxes on this bad boy?
-            unitNetPrice: total_amount,
-          })
-        })
-      } else {
-        const total_amount = item.content.unit_price * item.content.quantity
-
-        // Find the discount for current item and default to 0
-        const itemDiscount =
-          (itemDiscounts &&
-            itemDiscounts.find((el) => el.lineItem._id === item._id)) ||
-          0
-
-        // Withdraw discount from the total item amount
-        const total_discount_amount = total_amount - itemDiscount
+      const total_amount = item.unit_price * item.quantity
+      // Find the discount for current item and default to 0
+      const itemDiscount =
+        (
+          itemDiscounts &&
+          itemDiscounts.find((el) => el.lineItem.id === item.id)
+        )?.amount || 0
+
+      // Withdraw discount from the total item amount
+      const total_discounted_amount = total_amount - itemDiscount
+
+      order_lines.push({
+        lineNumber: order_lines.length + 1,
+        sortKey: 1,
+        unit: {
+          unitNumber: this.options_.unit_number,
+        },
+        product: {
+          productNumber: item.variant.sku,
+        },
+        quantity: item.quantity,
+        // Do we include taxes on this bad boy?
+        unitNetPrice: total_discounted_amount / 100,
+      })
+    })
 
+    if (!order.discounts.some(({ rule }) => rule.type === "free_shipping")) {
+      for (const shipping_method of order.shipping_methods) {
         order_lines.push({
```

Comment:
If you have any suggestions as to how shipping could be added in a more graceful maner I'm open to suggestions, I dont particularly like this

Reply:
Could it make sense to just go with `cart.shipping_total` instead of mapping over each method?

Reply:
done :)